model/task_lifecycle.go

```diff
@@ -1418,7 +1422,9 @@ func UpdateVersionAndPatchStatusForBuilds(buildIds []string) error {
 		return errors.Wrapf(err, "fetching builds")
 	}
 
-	versionsToUpdate := make(map[string]string)
+	// Maintain a list of builds for each version because we may
+	// be updating many builds for the same version.
+	versionsToUpdate := make(map[string][]string)
```

Comment:
nit: Maybe versionBuildsMap now instead to make it clear what the mapping was? The comment confused me at first since I didn't know where builds became relevant

Reply:
Sure!